src/wallet/wallet_puzzle_store.py

```diff
@@ -252,3 +293,20 @@ async def get_unused_derivation_path(self) -> Optional[uint32]:
             return uint32(row[0])
 
         return None
+
+    async def get_unused_derivation_path_for_wallet(
+        self, wallet_id: int
+    ) -> Optional[uint32]:
+        """
+        Returns the first unused derivation path by derivation_index.
+        """
+        cursor = await self.db_connection.execute(
```

Comment:
@matt-o-how This is the same issue that I mentioned to you before. This returns a derivation path with is unused for this wallet, but might be used by a different wallet. Therefore you might be reusing public keys. I think we should remove this method.